### PR TITLE
validation should match application reviews

### PR DESCRIPTION
**Pull request by hjacobs**
* URLs must start with `https://`
* git repos should not have the `.git` in the URL
* remove the specification type field (not used)

### CODE REVIEW

File: client/lib/application/src/application-form/application-form.jsx [hunk 1]
```diff
@@ -204,7 +204,8 @@ class ApplicationForm extends React.Component {
                         </div>
                         <div className='form-group'>
                             <label htmlFor='name'>Name</label>
-                            <small>The full name of your application.</small>
+                            <small>The full name of your application.
```
Comment (leviferreira) [inline client/lib/application/src/application-form/application-form.jsx]: 🎉 👏

**Comment by leviferreira** [comment]
👏

**Comment by leviferreira** [comment]
👍

File: client/lib/application/src/application-form/application-form.jsx [hunk 2]
```diff
@@ -246,25 +248,27 @@ class ApplicationForm extends React.Component {
...
+                            <small>Where you manage your source code.
+                            The application SHOULD have a link to the main source code repository (git repo).</small>
                             <input
-                                placeholder='https://github.com/zalando-stups/pierone.git'
+                                placeholder='https://github.com/zalando-stups/pierone'
                                 id='scm_url'
                                 value={app.scm_url}
-                                pattern='^https?://.*'
-                                title='Please provide an URL with http or https.'
+                                pattern='^https://.*'
```
Comment (mikkeloscar) [inline client/lib/application/src/application-form/application-form.jsx]: Can we have the full pattern? `^https://<host>/<org>/<repo>` ? Something like `https://[^/]+/[^/]+/[^/]+` is enough I guess?

Comment (hjacobs) [reply client/lib/application/src/application-form/application-form.jsx]: done

**Comment by leviferreira** [comment]
👍

**Comment by mikkeloscar** [comment]
:+1:

**Comment by leviferreira** [comment]
👍

**Comment by hjacobs** [comment]
:+1:

**Pull request merged**